src/Adapter/Shop/CommandHandler/UploadLogosHandler.php

```diff
@@ -70,72 +75,102 @@ public function __construct(
 
-        if (null !== $command->getUploadedHeaderLogo()) {
-            $this->uploadHeaderLogo($command->getUploadedHeaderLogo());
-        }
-
-        if (null !== $command->getUploadedMailLogo()) {
-            $this->uploadMailLogo($command->getUploadedMailLogo());
-        }
-
-        if (null !== $command->getUploadedInvoiceLogo()) {
-            $this->uploadInvoiceLogo($command->getUploadedInvoiceLogo());
-        }
-
-        if (null !== $command->getUploadedFavicon()) {
-            $this->uploadFavicon($command->getUploadedFavicon());
+        try {
+            if (null !== $command->getUploadedHeaderLogo()) {
+                $this->uploadHeaderLogo($command->getUploadedHeaderLogo());
+            }
+
+            if (null !== $command->getUploadedMailLogo()) {
+                $this->uploadMailLogo($command->getUploadedMailLogo());
+            }
+
+            if (null !== $command->getUploadedInvoiceLogo()) {
+                $this->uploadInvoiceLogo($command->getUploadedInvoiceLogo());
+            }
+
+            if (null !== $command->getUploadedFavicon()) {
+                $this->uploadFavicon($command->getUploadedFavicon());
+            }
+        } catch (PrestaShopException $exception) {
+            throw new ShopException(
+                'An unexpected error occurred when uploading image',
+                0,
+                $exception
+            );
         }
 
         $this->hookDispatcher->dispatchWithParameters('actionAdminThemesControllerUpdate_optionsAfter');
     }
 
     /**
      * @param UploadedFile $uploadedFile
+     *
+     * @throws ShopConstraintException
      */
     private function uploadHeaderLogo(UploadedFile $uploadedFile)
     {
-        $this->setUploadedFileToBeCompatibleWithLegacyUploader('PS_LOGO', $uploadedFile);
+        $legacyFileName = 'PS_LOGO';
+        $file = $this->setUploadedFileToBeCompatibleWithLegacyUploader($legacyFileName, $uploadedFile);
+        $this->assertIsValidImage($file[$legacyFileName]);
+
         $this->logoUploader->updateHeader();
     }
 
     /**
      * @param UploadedFile $uploadedFile
+     *
+     * @throws ShopConstraintException
      */
     private function uploadMailLogo(UploadedFile $uploadedFile)
     {
-        $this->setUploadedFileToBeCompatibleWithLegacyUploader('PS_LOGO_MAIL', $uploadedFile);
+        $legacyFileName = 'PS_LOGO_MAIL';
+        $file = $this->setUploadedFileToBeCompatibleWithLegacyUploader($legacyFileName, $uploadedFile);
+        $this->assertIsValidImage($file[$legacyFileName]);
 
         $this->logoUploader->updateMail();
     }
 
     /**
      * @param UploadedFile $uploadedHeaderLogo
+     *
+     * @throws ShopConstraintException
      */
     private function uploadInvoiceLogo(UploadedFile $uploadedHeaderLogo)
     {
-        $this->setUploadedFileToBeCompatibleWithLegacyUploader('PS_LOGO_INVOICE', $uploadedHeaderLogo);
+        $legacyFileName = 'PS_LOGO_INVOICE';
```

Comment:
could there be constants of a specific new class storing only the constants? :+1: